source/production/F0.Minesweeper.Components/Logic/Cell/CellVisualisationManager.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using F0.Minesweeper.Components.Abstractions;
+using F0.Minesweeper.Components.Abstractions.Enums;
+
+namespace F0.Minesweeper.Components.Logic.Cell
+{
+	internal class CellVisualisationManager : ICellVisualisationManager
+	{
+		private static readonly Dictionary<CellStatusType, (char? content, string cssClass)> visualisationTemplates = InitializeTranslations();
+		public CellVisualisation GetVisualisation(CellStatusType cellStatus) => GetVisualisation(cellStatus, null);
+		public CellVisualisation GetVisualisation(CellStatusType cellStatus, byte? adjacentMineCount)
+		{
+			(char? content, string cssClass) template = GetVisualisationTemplateOrDefault(cellStatus);
+
+			string cssClass = string.Format(CultureInfo.InvariantCulture, template.cssClass, adjacentMineCount);
+
+			if(template.content is null && adjacentMineCount is null)
+			{
+				throw new ArgumentNullException(nameof(adjacentMineCount));
+			}
+
+			char content = template.content ?? GetAdjacentMineCountContent(adjacentMineCount!.Value);
+
+			return new(content, cssClass);
+		}
+
+		private static char GetAdjacentMineCountContent(byte adjacentMineCount)
+		{
+			return adjacentMineCount switch
+			{
+				0 => ' ',
+				_ => adjacentMineCount.ToString(CultureInfo.InvariantCulture)[0]
```

Comment:
Hmm ... is there a more efficient way to convert the `byte` to `char` directly, rather than instantiating a new "throw-away" `string`? Would e.g. [Convert.ToChar](https://docs.microsoft.com/en-us/dotnet/api/system.convert.tochar) still yield the same result?

Reply:
At the time I implemented this I didn't find any better way. 
`Convert.ToChar` does not work since it converts the `byte` value to its unicode equivalent which would be `1 == ☺`, `2 == ☻`, `3 == ♥`, ...

And since I'm looking at this now again I decided I was stupid and I'll use something akin to `Convert.ToChar(48 + adjacentMineCount)` since it's measurably better.

![image](https://user-images.githubusercontent.com/16714208/148804717-4d53acb1-39ca-4c5d-8018-7ed7d985847a.png)

Reply:
Nice!
About `Benchy`: I see that you also see Nick Chapsas' videos 😉.

Reply:
Damn it! Next time I'll use `Benchö`. That's surely not a character found on a British keyboard 😉